### PR TITLE
Use clang-format for js code.

### PR DESCRIPTION
**Pull request by msbaines**
## Motivation and Context

We would like to use clang-format for js.

## How Has This Been Tested

Ran clang-format on js code and verified that running clang-format on C++ code caused no diff.

## Types of changes

- [X] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -27,5 +27,5 @@ repos:
       name: Run clang-format
       entry: clang-format -i -style=file
       types: [text]
-      files: '.*\.cpp|.*\.h'
+      files: '.*\.cpp|.*\.h|.*\.js'
```
Comment (erikwijmans) [inline .pre-commit-config.yaml]:
```suggestion
      files: '.*\.cpp|.*\.h|.*\.js$'
```

File: src/esp/bindings_js/simenv_embind.js [hunk 2]
```diff
@@ -11,7 +11,8 @@ class SimEnv {
    * @param {number} agentId - default agent id
    */
   constructor(config, episode, agentId) {
-    this.sim = new Module.Simulator(config);;
+    this.sim = new Module.Simulator(config);
+    ;
```
Comment (erikwijmans) [inline src/esp/bindings_js/simenv_embind.js]:
```suggestion
```

File: src/esp/bindings_js/navigate.js [hunk 3]
```diff
@@ -12,26 +12,30 @@ class NavigateTask {
...
-    this.radarCtx = components.radar.getContext("2d");
+    components.canvas.onmousedown = e => {
+      this.handleMouseDown(e);
+    } this.radarCtx = components.radar.getContext('2d');
     this.actions = [
-      { name: 'moveForward', key: 'w' },
-      { name: 'lookLeft', key: 'a', },
-      { name: 'lookRight', key: 'd' },
-      { name: 'done', key: ' ' }
+      {name: 'moveForward', key: 'w'}, {
```
Comment (apsdehal) [inline src/esp/bindings_js/navigate.js]: This formatting doesn't look right. Maybe worth to take a look if there is a flag that needs to be enabled.

File: src/esp/bindings_js/navigate.js [hunk 4]
```diff
@@ -12,26 +12,30 @@ class NavigateTask {
...
-    let shape = this.sim.getObservationSpace("rgb").shape;
+    this.imageCtx = components.canvas.getContext('2d');
+    let shape = this.sim.getObservationSpace('rgb').shape;
     this.imageData = this.imageCtx.createImageData(shape.get(1), shape.get(0));
-    this.semanticCtx = components.semantic.getContext("2d");
-    shape = this.sim.getObservationSpace("semantic").shape;
-    this.semanticImageData = this.semanticCtx.createImageData(shape.get(1), shape.get(0));
+    this.semanticCtx = components.semantic.getContext('2d');
+    shape = this.sim.getObservationSpace('semantic').shape;
+    this.semanticImageData =
```
Comment (apsdehal) [inline src/esp/bindings_js/navigate.js]: Same here. Maybe we should move to black. Worth checking if there are options there.
Black would make this
```js
this.semanticImageData = this.semanticCtx.createImageData(
    shape.get(1), shape.get(0)
)
```
which looks much nicer.

Comment (erikwijmans) [reply src/esp/bindings_js/navigate.js]: Does black work on javascript?  I thought it was python only

Comment (apsdehal) [reply src/esp/bindings_js/navigate.js]: Oh yeah. Sorry, working too much with python. There is a way to make it work with JavaScript but it is not worth it. In my mind, I was actually talking about eslint.

Comment (apsdehal) [reply src/esp/bindings_js/navigate.js]: ESLint is very nice in doing all this and very customizable.

Comment (msbaines) [reply src/esp/bindings_js/navigate.js]: Does black support javascript?

I wasn't able to find any option to fix this.

File: src/esp/bindings_js/navigate.js (hunk 3 above)
Comment (msbaines) [reply src/esp/bindings_js/navigate.js]: I was able to fix this a little bit.

**Comment by msbaines** [comment]
Abandoning this change. Will investigate ESLint instead since it is more widely used and clang-format has issues: weird formatting in cases and one instance where it broke code (closure).

**Pull request closed without merging**